yt_dlp/utils.py

```diff
@@ -2178,8 +2178,13 @@ class locked_file(object):
     _closed = False
 
     def __init__(self, filename, mode, block=True, encoding=None):
-        assert mode in ['r', 'rb', 'a', 'ab', 'w', 'wb']
-        self.f = io.open(filename, mode, encoding=encoding)
+        if mode not in ['r', 'rb', 'a', 'ab', 'w', 'wb']:
+            raise NotImplementedError(mode)
+        try:
+            fd = os.open(filename, os.O_WRONLY | os.O_CREAT | os.O_CLOEXEC)  # do not pass O_TRUNC
```

Comment:
where did u get these flags? They seem different from what io.open uses https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Modules/_io/fileio.c#L207

Reply:
I used strace. Typical line (before this PR) is:
```openat(AT_FDCWD</del/_yt_dlp_test_1646721829>, "youtube__Gwo3pEH7hUE.f160.mp4.part", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0666)```

If there is an inconsistency regarding fileio.c, I don't see it.